### PR TITLE
Add deckURI to NodeExecutionData

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Add deckURI to NodeExecutionData, so that Flyteconsole can retrieve the pre-signed deck URI and read the file. 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2175

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/data/implementations/aws_remote_url.go
Comment (EngHabu) [inline pkg/data/implementations/aws_remote_url.go:31]: I want to deprecate this entire file and use the new stdlib's storage Api to get presigned URLs... Is it too much to do as part of this PR?

Comment (pingsutw) [reply pkg/data/implementations/aws_remote_url.go:31]: ok, that makes sense. I think it's just a small change, will update this PR.

File: go.mod
Comment (katrogan) [inline go.mod:204]: friendly reminder to not merge this until after the idl version is updated :)

Comment (pingsutw) [reply go.mod]: updated it

File: dataproxy/service.go
Comment (katrogan) [inline dataproxy/service.go:95]: delete?

Comment (pingsutw) [reply dataproxy/service.go]: sorry, deleted it

**Comment by pingsutw** [comment]
cc @EngHabu @katrogan one question, I'm able to get a signed URL when using [flytekit](https://github.com/flyteorg/flytekit/pull/1034/files#diff-92ac8c8ecea8f27c4f68f8b66b39b54292b8a5de27a5a628d0eeed70690f2024R1008-R1020), but it doesn't work when I use curl. Anything I missed in this PR?

```bash
(base) ➜  ~ curl http://34.201.131.50:30081//api/v1/dataproxy/artifact_urn              
Not Found
(base) ➜  ~ curl http://34.201.131.50:30081//api/v1/dataproxy/artifact_urn\?native_url\=deck.html
Not Found
```

**Comment by katrogan** [comment]
@pingsutw why are there two slashes before api in the path?

**Comment by pingsutw** [comment]
Fixed it, we should register the data proxy handler in the HTTP server. https://github.com/flyteorg/flyteadmin/pull/413/files#diff-4ce88803abf59984cddcf6ab336c97b3231b0cb87f90e07f2f42e622e330f325R197-R201

File: dataproxy/service.go
Comment (pmahindrakar-oss) [inline dataproxy/service.go:104]: Can we move this to ValidateCreateDownloadLocationRequest

File: dataproxy/service.go
Comment (pmahindrakar-oss) [inline dataproxy/service.go:109]: Should we validate the NativeUrl aswell

File: dataproxy/service.go
Comment (pmahindrakar-oss) [inline dataproxy/service.go:120]: Is this valid  that we can accurately say here . The library sets this time for signed url if its google 
https://github.com/flyteorg/stow/blob/9c3f5f9ea24966206f0b2edc608da6ca803420e3/google/container.go#L55

**Review by pmahindrakar-oss — APPROVED** [review]

File: dataproxy/service.go
Comment (pingsutw) [reply dataproxy/service.go:98]: done

File: dataproxy/service.go
Comment (pingsutw) [reply dataproxy/service.go:109]: it's not accurate, but I think it's okay for now because we can't get the expiry time from [SignedURLResponse](https://github.com/flyteorg/flytestdlib/blob/d0a5767d30272c065a075a283fd2ba563834baca/storage/storage.go#L56-L58)
I can add it in a separate PR.

**Review by pmahindrakar-oss — APPROVED** [review]

**Pull request merged**
